app/src/main/java/org/piwigo/ui/login/LoginViewModel.java

```diff
@@ -95,15 +95,16 @@ void onLoginClick(FABProgressCircle fabCircle) {
 
         if (!siteValid)
             return;
-        fabCircle.show();
+        if (fabCircle != null)
```

Comment:
why is it allowed to call onLoginClick with a null parameter? I think it would be good to document this in a javadoc comment for that method. - I know so far it was also not documented but the "Nullable" property is helpful info for a caller, if it is explained what happens in the null case. Maybe we could even think of adding formal @Nullable annotations.

Reply:
ah, down there it looks like you add it for testing... This would be a "not so nice" workaround, as the test now executes different code then the real application, and we would never be able to reach a full test coverage.

Again, this is not a big deal and yes, I still count working code more than 100 nice unit test cases. But we have a bad test coverage already and we should try to improve, and I personally are not a big help here and it is a hard job for me, as I am really not experienced with testing. If you want to learn a good lesson it would be very appreciated if you could also do a work package on UnitTesting.

Reply:
We do have a bad coverage and I also think that having some unit testing is important, but yet we really need to keep going on getting to the 1.0 release milestone. 

Writing tests is not something fun for sure; but I’ll do it on time..

We need to keep in mind that people are waiting after us to release something reliable and as fast as possible.. so yet this is not our main focus and @plg should agree with me about this..

Reply:
I also agree, as written above... In this case, maybe it would be good to place a TODO into the code and/or create an issue assigned to a later milestone to document, that this @Nullable fabCircle is just a workaround to simplify unittest for now.

Reply:
Done in #129